porch/apiserver/pkg/registry/porch/background.go

```diff
@@ -56,15 +61,19 @@ func (b *background) run(ctx context.Context) {
 		}
 	}()
 
+	reconnect := newBackoffTimer(minReconnectDelay, maxReconnectDelay)
+	defer reconnect.Stop()
+
 	// Start ticker
 	ticker := time.NewTicker(1 * time.Minute)
 	defer ticker.Stop()
 
 loop:
 	for {
-		if watcher == nil {
+		select {
+		case <-reconnect.channel():
 			var err error
-			klog.Infof("Starting watch ... ")
+			klog.Info("Starting watch ... ")
```

Comment:
FYI I personally always use Infof, because the much more common failure is forgetting to use `f`, and I don't think the cost of Info vs Infof should be noticeable (though I haven't benchmarked)

Reply:
:) sg. putting it back.